unified_planning/environment.py

```diff
@@ -30,11 +31,22 @@ def __init__(self):
         import unified_planning.solvers
         import unified_planning.walkers
         self._type_manager = unified_planning.model.TypeManager()
-        self._factory = unified_planning.solvers.Factory()
+        self._factory = unified_planning.solvers.Factory(self)
         self._tc = unified_planning.walkers.TypeChecker(self)
         self._expression_manager = unified_planning.model.ExpressionManager(self)
         self._free_vars_oracle = unified_planning.model.FreeVarsOracle()
+        self._credits_stream: Optional[IO[str]] = sys.stdout
 
+    def __getstate__(self):
```

Comment:
write a couple of lines of comment to explain why this is needed, otherwise tomorrow we'll forget!